### PR TITLE
RGW: fix issue in observability over the persistent topics queue

### PR DESCRIPTION
**Pull request by AliMasarweh**
RGW: fix issue in observability over the persistent topics queue
when releasing entires, we don't decrease them in the urgent data

 Very recent bug, introduced in https://github.com/ceph/ceph/pull/52087

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/cls/queue/cls_queue_ops.h [hunk 1]
```diff
@@ -100,18 +100,21 @@ WRITE_CLASS_ENCODER(cls_queue_list_ret)
...
     encode(end_marker, bl);
+    encode(entries_to_remove, bl);
     ENCODE_FINISH(bl);
   }
 
   void decode(ceph::buffer::list::const_iterator& bl) {
-    DECODE_START(1, bl);
+    DECODE_START(2, bl);
     decode(end_marker, bl);
+    decode(entries_to_remove, bl);
```
Comment (yuvalif) [inline src/cls/queue/cls_queue_ops.h:117]: you should protect decoding with version check

File: src/cls/queue/cls_queue_client.cc [hunk 2]
```diff
@@ -78,7 +78,7 @@ int cls_queue_list_entries(IoCtx& io_ctx, const string& oid, const string& marke
   return 0;
 }
 
-void cls_queue_remove_entries(ObjectWriteOperation& op, const string& end_marker)
+void cls_queue_remove_entries(ObjectWriteOperation &op, const string &end_marker)
```
Comment (yuvalif) [inline src/cls/queue/cls_queue_client.cc:81]: nit:
```suggestion
void cls_queue_remove_entries(ObjectWriteOperation& op, const string& end_marker)
```

File: src/test/cls_queue/test_cls_queue.cc [hunk 3]
```diff
@@ -159,7 +159,7 @@ TEST_F(TestClsQueue, Dequeue)
   ASSERT_EQ(0, ret);
   ASSERT_EQ(truncated, true);
   // remove up to end marker
-  cls_queue_remove_entries(op, end_marker); 
+  cls_queue_remove_entries(op, end_marker);
```
Comment (yuvalif) [inline src/test/cls_queue/test_cls_queue.cc:162]: there should be no change in this file

**Comment by yuvalif** [comment]
please also run the cls tests they don't run in `ninja check`, and they don't run in the "notificartions" teuthology suite.
to run them, start a vstart cluster and then run:
```
bin/ceph_test_cls_2pc_queue
```
from the "build" directory

**Comment by AliMasarweh** [comment]
jenkins test make check

**Comment by AliMasarweh** [comment]
jenkins test make check

**Comment by AliMasarweh** [comment]
jenkins test make check

**Comment by AliMasarweh** [comment]
jenkins test make check

**Comment by AliMasarweh** [comment]
jenkins test make check

**Comment by AliMasarweh** [comment]
jenkins test make check

**Comment by AliMasarweh** [comment]
jenkins test make check

**Comment by AliMasarweh** [comment]
jenkins test make check

**Comment by AliMasarweh** [comment]
jenkins test make check

**Comment by AliMasarweh** [comment]
jenkins test make check

**Comment by AliMasarweh** [comment]
jenkins test make check

**Pull request merged**